app/controllers/application_controller/buttons.rb

```diff
@@ -723,7 +723,7 @@ def group_set_form_vars
     end
     @edit[:new][:available_fields] =
       CustomButton.buttons_for(@sb[:applies_to_class])
-                  .select { |u| u.parent.nil? }
+                  .select { |u| u.custom_button_sets.blank? }
```

Comment:
As it does here:

https://github.com/ManageIQ/manageiq/pull/21241/files#diff-d2682fba6af6b6b1a39d6403125a8b0476f9b7144d3d558fb0959ef05619716fR28-R29

You should also use a `.includes(:custom_button_sets)` so this doesn't cause an N+1.